lib/puppet/provider/service/systemd.rb

```diff
@@ -27,6 +27,7 @@
   defaultfor 'os.name' => :ubuntu
   notdefaultfor 'os.name' => :ubuntu, 'os.release.major' => ["10.04", "12.04", "14.04", "14.10"] # These are using upstart
   defaultfor 'os.name' => :cumuluslinux, 'os.release.major' => ["3", "4"]
+  defaultfor 'os.name' => :gentoo
```

Comment:
the confine resolves to false and the provider is excluded.

Reply:
Sorry forgot to mention could you add a test like https://github.com/puppetlabs/puppet/blob/03b3df33deb2d792ffd1c0eed9721fdc1d4833ad/spec/unit/provider/service/systemd_spec.rb#L166-L171

Also if you want this in 7.x, please submit a backport PR to that branch (since it relies on legacy facts)

Reply:
added a test, let me know it that's fine for you. After it's merged I will raise a PR for the 7.x branch.